quickstartKotlin/src/main/java/com/twilio/video/quickstart/kotlin/CameraCapturerCompat.kt

```diff
@@ -0,0 +1,110 @@
+package com.twilio.video.quickstart.kotlin
+
+import android.content.Context
+import android.util.Log
+import android.util.Pair
+import com.twilio.video.Camera2Capturer
+import com.twilio.video.CameraCapturer
+import com.twilio.video.VideoCapturer
+import org.webrtc.Camera2Enumerator
+
+class CameraCapturerCompat(context: Context, cameraSource: CameraCapturer.CameraSource) {
+    private val TAG = "CameraCapturerCompat"
+
+    private var camera1Capturer: CameraCapturer? = null
+    private var camera2Capturer: Camera2Capturer? = null
+    private var frontCameraPair: Pair<CameraCapturer.CameraSource, String>? = null
+    private var backCameraPair: Pair<CameraCapturer.CameraSource, String>? = null
+    private val camera2Listener = object : Camera2Capturer.Listener {
+        override fun onFirstFrameAvailable() {
+            Log.i(TAG, "onFirstFrameAvailable")
+        }
+
+        override fun onCameraSwitched(newCameraId: String) {
+            Log.i(TAG, "onCameraSwitched: newCameraId = " + newCameraId)
+        }
+
+        override fun onError(camera2CapturerException: Camera2Capturer.Exception) {
+            Log.e(TAG, camera2CapturerException.message)
+        }
+    }
+
+    init {
+        if (Camera2Capturer.isSupported(context)) {
+            setCameraPairs(context)
+            camera2Capturer = Camera2Capturer(context,
+                    getCameraId(cameraSource),
+                    camera2Listener)
+        } else {
+            camera1Capturer = CameraCapturer(context, cameraSource)
+        }
+    }
+
+    fun getCameraSource(): CameraCapturer.CameraSource {
+        return if (usingCamera1()) {
+            camera1Capturer!!.cameraSource
+        } else {
+            getCameraSource(camera2Capturer!!.cameraId)
+        }
+    }
+
+    fun switchCamera() {
+        if (usingCamera1()) {
+            camera1Capturer!!.switchCamera()
+        } else {
+            val cameraSource = getCameraSource(camera2Capturer!!
+                    .cameraId)
+
+            if (cameraSource == CameraCapturer.CameraSource.FRONT_CAMERA) {
+                camera2Capturer!!.switchCamera(backCameraPair!!.second)
+            } else {
+                camera2Capturer!!.switchCamera(frontCameraPair!!.second)
+            }
+        }
+    }
+
+    /*
+     * This method is required because this class is not an implementation of VideoCapturer due to
+     * a shortcoming in the Video Android SDK.
+     */
+    fun getVideoCapturer(): VideoCapturer {
```

Comment:
Can be converted to single line [property](https://kotlinlang.org/docs/reference/properties.html#getters-and-setters)